doc/migration/v0.1.0-migration-guide.md

```diff
@@ -0,0 +1,278 @@
+    ```Go
+    // Watch for changes to the primary resource Memcached
+    err = c.Watch(&source.Kind{Type: &cachev1alpha1.Memcached{}}, &handler.EnqueueRequestForObject{})
+
+    // Watch for changes to the secondary resource AppService and enqueue reconcile requests for the owner Memcached
+    err = c.Watch(&source.Kind{Type: &appv1alpha1.AppService{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &cachev1alpha1.Memcached{},
+	})
+    ```
+-  Add a new controller to watch and reconcile the CR independently of the other CR.
+    ```sh
+    $ operator-sdk add controller --api-version=app.example.com/v1alpha1 --kind=AppService
+    ```
+    ```Go
+    // Watch for changes to the primary resource AppService
+    err = c.Watch(&source.Kind{Type: &appv1alpha1.AppService{}}, &handler.EnqueueRequestForObject{})
+    ```
+
+### Copy and modify reconcile code from pkg/stub/handler.go
+
+In a `v0.1.0` project the reconcile code is defined in the `Reconcile()` method of a controller's [Reconciler][reconciler-go-doc]. This is similar to the `Handle()` function in the older project. Note the difference in the arguments and return values:
+- `func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error)`
+- `func (h *Handler) Handle(ctx context.Context, event sdk.Event) error`
+
+Instead of receiving an sdk.Event (with the object), the Reconcile function receives a [Request][request-go-doc] (Name/Namespace key) to lookup the object. If an error is returned the Request will be retried. Otherwise the [Result][result-go-doc] can be explicitly configured to retry immediately or periodically after a timeout.
+
+Copy the code from the old project's `Handle()` function over the existing code in your controller's `Reconcile()` function.
+Be sure to keep the initial section in the `Reconcile()` code that looks up the object for the Request and checks to see if it's deleted.
+
+```Go
+import (
+    apierrors "k8s.io/apimachinery/pkg/api/errors"
+    cachev1alpha1 "github.com/example-inc/memcached-operator/pkg/apis/cache/v1alpha1"
+    ...
+)
+func (r *ReconcileMemcached) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	// Fetch the Memcached instance
+	instance := &cachev1alpha1.Memcached{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+    }
+
+    // Rest of your reconcile code goes here.
+    ...
+}
+```
+
+Change the following in your reconcile code:
+- Replace `return err` with `return reconcile.Result{}, err`
+- Replace the calls to the SDK client(Create, Update, Delete, Get, List) with the reconciler's client
+
```

Comment:
Do we want to add:
```md
- Replace `return nil` with `return reconcile.Result{}, nil`
```
Should we mention `RequeueAfter`? We allude to this possibility a few paragraphs up, but since we call out how to change `return err`, we may want to be more explicit for `return nil` as well?

Reply:
Yes I'll mention how to replace `return nil` as well.

And I'll add a small section on periodic reconcile and `RequeueAfter`. We've mentioned it in the user-guide but it isn't obvious how this replaces the way we used to do periodic reconcile in the SDK.